packages/main/src/DateRangePicker.js

```diff
@@ -84,6 +92,10 @@ class DateRangePicker extends DatePicker {
 		return [DatePicker.styles, DateRangePickerCss];
 	}
 
+	onAfterRendering() {
+		this._prevDelimiter = this._effectiveDelimiter;
```

Comment:
This causes unnecessary invalidation (as _prevDelimiter is entry in the metadata and you change its value in onAfterRendering, which starts entirely new invalidation cycle).
You can just declare the variable in the constructor, it will keep track the previous delimiter value:
```js
 this._prevDelimiter = null;
```
Then, in _splitValueByDelimiter:

```js
// as the first time this._prevDelimiter would be null,
// you can fallback to the effective as it used to be
const partsArray = value.split(this._prevDelimiter || this._effectiveDelimiter);

// after that, you sync the values
this._prevDelimiter = this._effectiveDelimiter;

```

Reply:
Done